### PR TITLE
add setUserData and getUserData utility functions

### PR DESCRIPTION
**Pull request by gka**
I tried running the `INSERT ... ON DUPLICATE INSERT` (aka mysql-upsert) as Sequelize `UserData.upsert` query but couldn't get it to work so I used the raw query instead.

### CODE REVIEW

File: utils/userData.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * a quick way to set or update a user setting in user_data
+ * @param {number} userId
+ * @param {string} key
+ * @param {string} value
+ */
+module.exports.setUserData = async function(userId, key, value) {
+    return ORM.db.query(
+        'INSERT INTO user_data(user_id, `key`, value) VALUES (:userId, :key, :value) ON DUPLICATE KEY UPDATE value = :value, stored_at = CURRENT_TIMESTAMP',
```
Comment (sto3psl) [inline utils/userData.js]: This looks a lot like `Model.upsert` in Sequelize.
https://sequelize.org/master/class/lib/model.js~Model.html#static-method-upsert

Comment (gka) [reply utils/userData.js]: yes, as I said in the initial PR comment, I tried using upsert but couldn't get it to work afte 30 minutes so I gave up

Comment (sto3psl) [reply utils/userData.js]: Sorry, I misread that. Shouldn't work when I still have this headache.

**Comment by gka** [comment]
added another function for removing userData entries, and also some tests

**Pull request merged**